Replicator/Replicator.cc

```diff
@@ -1307,4 +1295,36 @@ namespace litecore { namespace repl {
         }
     }
 
+    void Replicator::setMsgHandlerFor3_0_Client(Retained<blip::MessageIn> request)
+    {
+        if (_setMsgHandlerFor3_0_ClientDone) {
+            return;
+        } else {
+            _setMsgHandlerFor3_0_ClientDone = true;
+        }
+
+        if (_options->isActive()) {
+            return; // only deal with passive replicator.
+        }
+
+        if (request->intProperty(kCollectionProperty, kNotCollectionIndex)
+            != kNotCollectionIndex) {
+            return; // 3.0 message should not include the collection property
+        }
+
+        std::vector<C4CollectionSpec> v;
+        v.emplace_back(kC4DefaultCollectionSpec);
+        _options->rearrangeCollectionsFor3_0_Client();
+        DebugAssert(!_options->collectionAware());
+        DebugAssert(_options->workingCollectionCount() == 1);
+        if (!_options->collectionPath(0)) {
```

Comment:
What is this testing?  The log message doesn't seem to line up with the test.  Shouldn't this be compared to the default collection path rather than just `!`?

Reply:
when the default collection is not listed in the array of ReplicationCollection of ReplicatorParameters. rearrangeCollectionFor3_0_Client() will pull the default collection to the beginning of the array. If it's missing, nullslice will take that place.

Reply:
This doesn't prove that.  What if there is collection A, instead of the default collection, in position 0?

Reply:
We are following the general rule: if at index i, the collection is not specified by the passive replicator, we substitute null collection path. In this case, there is only one collection in the config, with index 0.

Reply:
I see, this is more context that will not make sense to a new reader without comments.  The key here seems to be that we have the assertion `!_options->collectionAware()`.  This means that we are forced into only having at most 1 collection, which can only be the default one or nothing.  See how just saying "the general rule that if the collection is not specified then we substitute null" doesn't tell the whole story?

Reply:
"forced into only having at most 1 collection" -- not "at most". 0 collection would be ruled out before arriving here. At this place, there is exactly one collection.